test/parallel/test-crypto-dh-stateless-async.js

```diff
@@ -0,0 +1,220 @@
+'use strict';
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const crypto = require('crypto');
+const { hasOpenSSL3 } = require('../common/crypto');
+
+assert.throws(() => crypto.diffieHellman(crypto.generateKeyPairSync('ec', { namedCurve: 'P-256' }), null), {
+  name: 'TypeError',
+  code: 'ERR_INVALID_ARG_TYPE',
+  message: 'The "callback" argument must be of type function. Received null'
+});
+
+function test({ publicKey: alicePublicKey, privateKey: alicePrivateKey },
+              { publicKey: bobPublicKey, privateKey: bobPrivateKey },
+              expectedValue) {
+  crypto.diffieHellman({
+    privateKey: alicePrivateKey,
+    publicKey: bobPublicKey
+  }, common.mustSucceed((buf1) => {
+    if (expectedValue !== undefined)
+      assert.deepStrictEqual(buf1, expectedValue);
+    crypto.diffieHellman({
+      privateKey: bobPrivateKey,
+      publicKey: alicePublicKey
+    }, common.mustSucceed((buf2) => {
+      assert.deepStrictEqual(buf1, buf2);
+    }));
+  }));
+}
+
+const alicePrivateKey = crypto.createPrivateKey({
+  key: '-----BEGIN PRIVATE KEY-----\n' +
+       'MIIBoQIBADCB1QYJKoZIhvcNAQMBMIHHAoHBAP//////////yQ/aoiFowjTExmKL\n' +
+       'gNwc0SkCTgiKZ8x0Agu+pjsTmyJRSgh5jjQE3e+VGbPNOkMbMCsKbfJfFDdP4TVt\n' +
+       'bVHCReSFtXZiXn7G9ExC6aY37WsL/1y29Aa37e44a/taiZ+lrp8kEXxLH+ZJKGZR\n' +
+       '7ORbPcIAfLihY78FmNpINhxV05ppFj+o/STPX4NlXSPco62WHGLzViCFUrue1SkH\n' +
+       'cJaWbWcMNU5KvJgE8XRsCMojcyf//////////wIBAgSBwwKBwEh82IAVnYNf0Kjb\n' +
+       'qYSImDFyg9sH6CJ0GzRK05e6hM3dOSClFYi4kbA7Pr7zyfdn2SH6wSlNS14Jyrtt\n' +
+       'HePrRSeYl1T+tk0AfrvaLmyM56F+9B3jwt/nzqr5YxmfVdXb2aQV53VS/mm3pB2H\n' +
+       'iIt9FmvFaaOVe2DupqSr6xzbf/zyON+WF5B5HNVOWXswgpgdUsCyygs98hKy/Xje\n' +
+       'TGzJUoWInW39t0YgMXenJrkS0m6wol8Rhxx81AGgELNV7EHZqg==\n' +
+       '-----END PRIVATE KEY-----',
+  format: 'pem'
+});
+const alicePublicKey = crypto.createPublicKey({
+  key: '-----BEGIN PUBLIC KEY-----\n' +
```

Comment:
Nit: FWIW, I think it'd be slightly better to use fixtures here. (Feel free to ignore.)